skopt/learning/gaussian_process/gpr.py

```diff
@@ -186,12 +186,13 @@ def fit(self, X, y):
         if self.kernel is None:
             self.kernel = ConstantKernel(1.0, constant_value_bounds="fixed") \
                           * RBF(1.0, length_scale_bounds="fixed")
-        if self.noise == "gaussian":
-            self.kernel = self.kernel + WhiteKernel()
-        elif self.noise:
-            self.kernel = self.kernel + WhiteKernel(
-                noise_level=self.noise, noise_level_bounds="fixed"
-            )
+        if self.noise and not _param_for_white_kernel_in_Sum(self.kernel)[0]:
```

Comment:
Isn't this the same as moving the whole thing into the constructor?

Reply:
The thing is `self.kernel` can be `None` in the constructor. Therefore, calling `_param_for_white_kernel_in_Sum(self.kernel)` will raise `NoneType` associated exceptions.

Will make some tests and add them to test_gpr.py.

Reply:
Added unit test.

The test fails with the old implementation, passes with the new one.